### PR TITLE
Explaining and calibrating epochs

### PR DESCRIPTION
**Pull request by ThomasArts**

### CODE REVIEW

File: Periodically-Syncing-HyperChains.md [hunk 1]
```diff
@@ -533,12 +533,78 @@ You can not withdraw funds if the deposited balance would go under `MINIMUM_STAK
...
+- The seed should be known before the actual schedule needs to be computed, otherwise the child chain is stuck
+- The stakers should actually do work, so only after the work is done, their rewards for doing the work is paid out.
+
+This leads to a design with of a staking cycle that consists of four distinct phases:
+
+1. **Staking Phase**: Participants register and adjust their stakes.
+   In the staking phase, we collect all stakes posted in the ongoing child epoch. The result at the Nth child epoch CE(n) is refered to as s(n).  The initial stake as configured for the child chain is s(0).
+
+2. **Leader Election Phase**: The system uses the state of the parent chain and the stakes recorded to generate a schedule for selecting leaders.
+   In the leader election phase we retrieve the last hash of a parent chain epoch to be used as a seed later on. For CE(n) we store the last blockhash of PE(n-2) as seed for later schedule computation.
```
Comment (hanssv) [inline Periodically-Syncing-HyperChains.md]: I think currently we use the _first_ block of parent epochs, but that is just a technicality.

File: Periodically-Syncing-HyperChains.md [hunk 2]
```diff
@@ -533,12 +533,78 @@ You can not withdraw funds if the deposited balance would go under `MINIMUM_STAK
...
+   Note that effectively at this moment we know the schedule for block production 2 epochs ahead. Among others, stakers now can be more attentive.
+3. **Block Production Phase**: Only validators meeting the minimum staking requirements are eligible to produce blocks.
+   The schedule for CE(n) block production is based upon the stake set in CE(n-4) and the seed from the last block of PE(n-2).
+4. **Payout Phase**: Rewards are distributed based on block production results.
+
+Note that each child epoch has all of these characteristics, viz. child epoch nine is the payout  phase from a cycle
```
Comment (hanssv) [inline Periodically-Syncing-HyperChains.md]:
```suggestion
Note that each child epoch has all of these characteristics, viz. child epoch nine is the payout  phase for a cycle
```

File: Periodically-Syncing-HyperChains.md (hunk 1 above)
Comment (ThomasArts) [reply Periodically-Syncing-HyperChains.md]: Good point.

**Review by hanssv — APPROVED** [review]

File: Periodically-Syncing-HyperChains.md [hunk 3]
```diff
@@ -533,12 +533,87 @@ You can not withdraw funds if the deposited balance would go under `MINIMUM_STAK
...
 #### Staking Cycle Structure
-A staking cycle consists of four distinct epochs:
 
-1. **Staking Epoch**: Participants register and adjust their stakes.
-2. **Leader Election Epoch**: The system uses the state of the parent chain and the stakes recorded to generate a schedule for selecting leaders.
-3. **Block Production Epoch**: Only validators meeting the minimum staking requirements are eligible to produce blocks.
-4. **Payout Epoch**: Rewards are distributed based on block production results.
+Each epoch needs a schedule of producers. These producers are randomly selected from registered stakers.
+We are concerned about the following:
+- To prevent stakers to influence the schedule, the staking distribution should be known before the seed of the random selection is known.
```
Comment (mitchelli) [inline Periodically-Syncing-HyperChains.md:539]: I would suggest, "from influencing" rather than "to influence"

**Pull request merged**